docs/assembly.rst

```diff
@@ -596,43 +596,43 @@ of their block is reached.
 Conventions in Solidity
```

Comment:
My main thought is that the title is "Conventions in Solidity", but there are actually two somewhat unrelated topics under the heading, I wonder if instead there should be two distinct titles.

1. High-order bit cleaning? (I'm not actually sure)
2. Memory management

Reply:
These are things you have to know about if you are mixing solidity and inline assembly.

Reply:
@chriseth How about this instead?

```suggestion
Considerations in Solidity
```

I'll need to fix the underline

Reply:
This section explains how the compiler uses certain resources and how it deals with smaller types. If you use the words "considerations" then it should rather be "considerations when using inline assembly in solidity", but maybe there is a better word? :)

I used "conventions" because this word is usually used in e.g. mathematical texts when it is explained how certain symbols are used and which implicit assumptions are made when doing so.

Reply:
OK, then! I think anyone using assembly kind of knows what they're doing, so let's leave as is.